File: media/src/main/java/com/smasher/media/SmasherApplication.java
package com.smasher.media;

import androidx.multidex.MultiDexApplication;

import com.smasher.zxing.activity.ZXingLibrary;
import com.squareup.leakcanary.LeakCanary;
import com.squareup.leakcanary.RefWatcher;

/**
 * @author matao
 * @date 2019/6/10
 */
public class SmasherApplication extends MultiDexApplication {

    private static RefWatcher sRefWatcher;

    @Override
    public void onCreate() {
        super.onCreate();
        ZXingLibrary.initDisplayOpinion(this);

        if (LeakCanary.isInAnalyzerProcess(this)) {
            return;
        }
        sRefWatcher = LeakCanary.install(this);
    }


    public static RefWatcher getRefWatcher() {
        return sRefWatcher;
    }
}

File: media/src/main/java/com/smasher/media/SmasherApplication.kt
//package com.smasher.media
//
//import android.content.Context
//import androidx.multidex.MultiDexApplication
//import com.smasher.zxing.activity.ZXingLibrary
//import leakcanary.LeakCanary
//import leakcanary.LeakSentry
//
///**
// * @author matao
// * @date 2019/6/10
// */
//class SmasherApplication : MultiDexApplication() {
//
//
//    override fun attachBaseContext(base: Context?) {
//        super.attachBaseContext(base)
//    }
//
//
//    override fun onCreate() {
//        super.onCreate()
//
//        LeakSentry.config = LeakSentry.config.copy(watchFragments = false)
//    }
//}